### DISCUSSION TITLE
How to crawl/scrap content that is SSG with getStaticProps

### DISCUSSION THREAD

**Discussion by damianfilipek81** [vercel/next.js#61998] [Help]
### Summary

Hello!
I have created website using next.js 14.1 with app directory. Let's say i have a root layout that looks like this: 

```tsx   
     <body className="min-h-screen bg-mainBackground flex flex-col">
        <Providers>
          <Header />
          <div className="container flex flex-col flex-grow">{children}</div>
          <Footer />
        </Providers>
      </body>
```

Root layout component is a server component. `Providers` one, naturally is a client component. Now i'm building my website and in the output of `pnpm build` i can see 

```
├ ● /[lng]                                6.92 kB         221 kB
├   ├ /en
├   └ /pl
```
```
○  (Static)   prerendered as static content
●  (SSG)      prerendered as static HTML (uses getStaticProps)
λ  (Dynamic)  server-rendered on demand using Node.js
```

So i'm sure that initial page is generated by SSG. The problem is that not a single one online crawler/scrapper can find content on my pages that are rendered in the `Providers` component. I can crawl and scrap thing that are outside of it, but almost every component needs context, which i have in the `Providers` component. I'm aware that we can pass server components to the client's one through the children prop, which im doing it right know, but i still cannot overcome this issue. 
I'm aware of the crawlers limitations for dynamic rendering but shouldn't it work if it's server-side generated pages?
What am i missing here? Is it normal behavior?

**Comment by skycloudnest** [comment]
I aam having the same problem!
